### DISCUSSION TITLE
Selecting DataColumns to be stored by Cognition.run

### DISCUSSION THREAD

**Discussion by mingjia1025** [jspsych/jsPsych#1104] [Q&A] [answered]
I'm using cognition.run to host my jsPsych experiment which automatically saves the data at end of each run. The problem is that it saves all the experiment-related  information by default (including many columns not relevant for my data analysis). Is it possible to select data columns I want it to save?

**Comment by jodeleeuw** [comment]
I'm not sure how cognition.run automates the data collection process. In a normal jsPsych experiment you could filter out irrelevant columns with [.ignore()](https://www.jspsych.org/core_library/jspsych-data/#ignore) before sending the data to the server, but I think cognition.run injects some code to handle this for you. I'm not sure how to intercept that data before it is saved. 

Maybe @javidalpe can chime in?

**Comment by javidalpe** [comment, marked as answer]
Hello @mingjia1025. And thanks @jodeleeuw for the mention.

I don't think you can do it on cognition.run, sorry. Is there any legal/technical problem about storing all data?


---
FYI

The app just calls jsPsych.data.get().csv() during the on_finish event and sends the data to the server, hehe.

I could add some kind of callback system for selecting the data before uploading it, but I refuse to add cognition-specific code to the platform. New specific code would require new mental models to understand what is jspysch code and what is cognition code, new documentation, new forums to report specific code problems.

**Reply by mingjia1025** [reply]
It's okay to store all the data for me but I still prefer to have a way of selecting specific rows and columns to be stored as an excel file as that would make analysis of my data much easier. Is there a way for you to tailor the app to my need without any cognition-specific code? Please give me a template of how to do it programmatically.

**Reply by kernel52** [reply]
Hi @javidalpe,
Most of the data, you can exclude one way or another (using on_trial_finish for example and deleting what's not needed), and I assume as the library evolves, it will be possible to prohibit some trials from inserting info in the data object, keeping the csv light. 

The issue might be with the extra columns in the data object that are "injected" before you launch the jsPsych.data.get().csv(). 

There is an option for storing the ip, but what about the rest ? 
recorded_at | user_agent | device | browser | browser_version | platform | platform_version

shouldn't this be a user's choice to have or not? :)
but again, this is just a suggestion, the tool is very useful and life is too short to complain
